src/state.rs

```diff
@@ -44,8 +43,8 @@ enum DkgCurrentState {
     WaitingForMoreParts,
 }
 
-impl<R: bls::rand::RngCore + Clone> DkgState<R> {
-    pub fn new(
+impl DkgState {
+    pub fn new<R: bls::rand::RngCore>(
```

Comment:
Don't ask for  a mutable reference to `R` in methods. `rand` has a default implementation that says `&mut RngCore is an RngCore`:

https://docs.rs/rand/latest/rand/trait.RngCore.html#impl-RngCore-for-%26%27a%20mut%20R
![mutrng-is-rng](https://user-images.githubusercontent.com/1832378/188512721-cebeeb62-49e0-4f39-bacf-016450971301.png)


This means users can pass in `&mut RngCore` even if you asked for `RngCore`